### DISCUSSION TITLE
Changing PC and delay while stepping

### DISCUSSION THREAD

**Discussion by coder-au** [prb28/vscode-amiga-assembly#251] [Q&A]
Hey love the extension. Makes life so much easier/fun! The debugger integration is brilliant. Have been running with WinUAE now for several weeks and solved many longer standing tricky bugs in my code base. Super helpful. Thank you for your hard work on this.

I have a bit of a feature request/question. Is it possible to change the address of the program counter? Am running v1.7.1 Pre-Release.

Also a side, fairly often stepping through code will lag for both over or into. I see a lot of text hitting the Debug Console. I thought it might be related. If I am not paying attention and step too many times stepping will lock up and I need to kill the process.

**Comment by prb28** [comment]
In the lauch.json file do you have the trace=true option?
In this case try to set it ton false.

**Reply by coder-au** [reply]
Hi, thanks for response. I did have trace enabled, disabled now. Still with the lag when stepping at times though. When it occurs I have just noticed the cpu field in the call stack shows a waiting icon.

How about for the PC, is there any way to change that while stepping? (WinUAE)

**Reply by prb28** [reply]
You can add a breakpoint and let it run until it's hit.

**Reply by coder-au** [reply]
> You can add a breakpoint and let it run until it's hit.

Sure thanks! but can I change it without executing the code like would happen if I let it run to a breakpoint. If I want to move the PC back or forward, or to another address entirely. Is it possible?

**Reply by prb28** [reply]
I don't think that jumping to a pc is possible (but I never tried, you can try it by setting the pc value). You can add to your code some conditional jump function based on a register to help you achieve that.
Did you try to use the conditional breakpoints? Maybe it can help.

**Reply by coder-au** [reply]
> but I never tried, you can try it by setting the pc value

Editing the address directly in the registers list results in "Unknown Register" error. Nice red box shows up and no change is made to the PC sadly. On 68k the PC is not writable if I remember correctly, but maybe since running through emulation WinUAE might provide some means to set the address directly that could be "hooked up". Dragging that little yellow arrow in source code showing current line when debugging would be great for productivity in some cases.
